### ISSUE TITLE
Reposition characters that override the soft-hyphen

### ISSUE THREAD

**Issue by laicasaane** [BenTalagan/glaemscribe#5]
Please reposition any character that overrides the soft-hyphen (U+00AD). Because soft-hyphens are always hidden in modern text processors, then characters at this position are hidden also.
Fonts affected: Tengwar Annatar, Tengwar Parmaite, Tengwar Elfica, Tengwar Sindarin (also Quenya).

Currently Tengwar Eldamar uses the position U+20AC instead.

**Comment by BenTalagan** [comment]
Hello, and thanks for noticing!

Actually it is already on my todo list. I've realised the problem recently too, because I'm working on a "pure spelling raw tengwar" mode, that would allow to enter tengwar by spelling their name (ex: noldo-a-malta-ara-a-romen-palatal-e for namárië). This mode would allow complete control on the input, and would also be available in all tengwar modes with an escape sequence to toggle that "raw spelling" mode. Working on that mode made that soft hyphen problem appear to me, the concerned character is a lower 'a theta' and it was almost not used in the currently implemented modes. There is already a solution for Elfica, since it proposes other mappings for all tehtar so that one will be easy (as well as Eldamar, as you noticed). 

Unfortunately, a patch is needed for all other fonts ; each time, it's a bit of painful work to refactor the fonts. The versions of the fonts used by Glaemscribe are slowly diverging from their original versions from their original authors by a renewed sum of minor patches, and it is a bit embarrassing since they slowly get more and more desynchronized. (I really praise the Free Tenwar Project by J. Mach Wust and Johan Winge for their work on the next generation of intelligent fonts. It will help solve a lot problems).

Thus, please be patient for that fix for U+00AD, I'm currently under heavy load professionally and will try to patch the fonts, the charsets and make the raw mode available in a few weeks from now :)

**Comment by laicasaane** [comment]
I don't know how's about the other authors, but I have had a word from Johan Winge himself states that he won't find it a problem if we make some changes to the layout of his fonts to improve them. Months ago I've asked him a permission for modifying Tengwar Annatar. I intended to develop a semi-Unicode layout which contains all existing characters without using OpenType or Graphite technology. It's just some kind of Dan Smith layout, but this time the Tengwar characters will be relocated in PUA blocks following the Unicode layout for Tengwar. For that they won't interfere with other standard characters anymore. However, that idea is still mainly in my mind. It's hard to produce an example font since I'm totally not good at font things.

Well, I'm surely patient to wait for your patch since it's not very urgent now. (I can still use Tengwar Eldamar instead.)

P/S: Many thanks for this Glaemscribe. It did help me through a lot of troubles making the Tengwar mode for my mother tongue. I fully appreciate your work!

**Comment by BenTalagan** [comment]
Thank you very much for your kind words! This is truly heart-warming.

I could find some time yesterday to work actively on your issue and have managed to rework a bit the tengwar charsets but also patch the fonts currently used by Glaemscribe (Annatar Regular/Bold/Italic, Eldamar, Parmaite and Sindarin - Elfica was ok thanks to its mapping redundancy); I've also implemented a first version (not totally complete) of the 'raw inlined tengwar' mode, and added a system to 'call' that mode within any tengwar mode (by enclosing spelled tengwar inside blocks like this : {{ numen/a/malta/o</palatal  etc etc }} ; it gives the end-user a full control on the output when the mode logic is not sufficient. 

Doing this, I have encountered other font issues (concerning the 'vaia' tengwa, which is quite rare) which was mapped on a problematic hangul character. For the moment, my strategy is to relocate/add missing or problematic characters to Enrique Mombello's mapping from Elfica which is the only 'old gen' font that I haven't had to patch yet. Elfica has some redundancy in its mapping, and this proves quite helpful. Unifying all legacy fonts to a unicode mapping would be a great thing (especially if following the Free Tengwar Font & Everson mapping), but I don't know how hard it would be. Relocating chars is easy with FontForge, but maybe it would break kerning information & raise other problems ? I can't way for the release of new Tengwar OpenType fonts especially to get rid of tricky tehtar + sarince + consonant modifications combinations.

For the moment, I've been releasing patched fonts for Glaemscribe on the Glaemscrafu website [here](https://www.jrrvf.com/glaemscrafu/french/about-transcriptions.html) with the benediction of Måns Björkman too. The only exception is the patched DS 'Tengwar Sindarin' font which I don't release but is still available in Glaemscribe's user interface - but it's here more for demonstration/nostalgie purpose. Dan Smith fonts start to lack a lot of features and they require more work (and they were asked for removal by the Tolkien Estate), and generally speaking I'm happier with Annatar or Eldamar. I still hesitate to release Glaemscribe's patched fonts on github, for proprietary reasons. They're in the same time tangential but part of the project, but are the propriety of their authors and respecting the work of everyone (starting from JRRT) is my prior concern.

Anyway, if you have interesting things to share, I'd love to see your mode implementation when you feel it's showable! I'm to prepare the new release for this week-end, it still needs a bit of testing and documentation (for the inlined raw tengwar feature) but I'm confident.

Thank you very much for your interest!
